mypy/checker.py

```diff
@@ -7519,6 +7556,55 @@ def has_valid_attribute(self, typ: Type, name: str) -> bool:
     def get_expression_type(self, node: Expression, type_context: Type | None = None) -> Type:
         return self.expr_checker.accept(node, type_context=type_context)
 
+    def get_deprecation_warning(self, decorators: Iterable[Expression]) -> str | None:
+        for decorator in decorators:
+            if (
+                isinstance(decorator, CallExpr)
+                and isinstance(callee := decorator.callee, NameExpr)
+                and (
+                    (callee.fullname in DEPRECATED_TYPE_NAMES)
+                    or (not callee.fullname and callee.name == "deprecated")
```

Comment:
Does this mean that an unresolved decorator `@deprecated` gets treated like the PEP 702 deprecated?

Reply:
Yes, I am afraid so.  

I wrote this line when working on `testDeprecatedProperty`.  In `TypeChecker._visit_overloaded_func_def`, `defn.items[1].callee.fullname == ""`.  Is this a Mypy bug or am I missing something?

I will remove this line and let the test fail for now.

Reply:
It seems the semantic analyser needed a little tweak to analyse the additional decorators of the non-getter overload parts.   `testDeprecatedProperty` passes again.  Other decorators named `deprecated` should not become confused with those of  `warnings` and `typing_extensions` anymore.